### ISSUE TITLE
Setting marionette to true in Selenium::WebDriver::Remote::Capabilities sets Webdriver to local

### ISSUE THREAD

**Issue by Marcelovk** [SeleniumHQ/selenium#2281]
## Meta -
OS:  Linux
Selenium Version:  2.53.1 (selenium-webdriver Ruby gem)
Browser:  Firefox 46

## Expected Behavior 

-  A new session starts in selenium grid

## Actual Behavior 

- Test fails stating that there is no firefox binary in the same machine

## Steps to reproduce -

Just try to use marionette on a Selenium grid using ruby:

```
                caps = Selenium::WebDriver::Remote::Capabilities.firefox(:marionette => true)
                Selenium::WebDriver.for(:remote,
                                        :url => "http://172.17.0.5:4444/wd/hub",
                                        :desired_capabilities => caps)
```

The call never gets to selenium-server, instead it fails with the following message:

Selenium::WebDriver::Error::WebDriverError:
       Could not find Firefox binary (os=linux). Make sure Firefox is installed or set the path manually with Selenium::WebDriver::Firefox::Binary.path=

**Comment by titusfortner** [comment]
Does this work with 2.53.0? 2.53.1 switched the path finding to a `Dir.glob` functionality, and apparently it behaves differently on each platform. I'm on a road trip without access to my linux machine, I'll try to get a VM to look at this soon.

In the meantime, if you want to run this and let me know what you see I might be able to figure out what is going on faster.
```
paths = ENV['PATH'].split(File::PATH_SEPARATOR)
binary_name = 'firefox'
paths.each do |path|
  full_path = File.join(path, binary_name)
  puts full_path
  exe = Dir.glob(full_path).first
  next unless exe
  puts exe if File.executable?(exe)
  break
end
```

**Comment by Marcelovk** [comment]
I did not run the code, as I do not have Firefox installed in the same machine that runs the test against the selenium-grid. The issue is also present in 2.53.0

I do not have much experience with Ruby, however I traced down the issue in https://github.com/SeleniumHQ/selenium/blob/master/rb/lib/selenium/webdriver/remote/w3c_bridge.rb#L80

It seems that Firefox.binary in this line (
https://github.com/SeleniumHQ/selenium/blob/master/rb/lib/selenium/webdriver/firefox/binary.rb#L156) needs a file to check the version, which is not the case in my test machine (I am using a docker container to test and run the tests). Commenting out that comparision makes it at least to connect in selenium-server (then causes other errors that are out of scope of this issue)

**Comment by titusfortner** [comment]
Yup, you're right that doesn't work when running on a remote machine.
Fixed in selenium-webdriver-2.53.3.gem
